File: frontend/ProjectManagement.Core/Models/UpdateProjectRequest.cs
namespace ProjectManagement.Core.Models;

/// <summary>
/// Request to update an existing project.
/// Only non-null fields will be updated.
/// </summary>
public sealed record UpdateProjectRequest
{
    /// <summary>
    /// ID of the project to update.
    /// </summary>
    public required Guid ProjectId { get; init; }

    /// <summary>
    /// Expected version for optimistic concurrency.
    /// </summary>
    public required int ExpectedVersion { get; init; }

    /// <summary>
    /// New title (null to keep current).
    /// </summary>
    public string? Title { get; init; }

    /// <summary>
    /// New description (null to keep current).
    /// </summary>
    public string? Description { get; init; }

    /// <summary>
    /// New status (null to keep current).
    /// </summary>
    public ProjectStatus? Status { get; init; }
}

File: backend/pm-server/src/api/projects/update_project_request.rs
use serde::Deserialize;

#[derive(Debug, Deserialize)]
pub struct UpdateProjectRequest {
    #[serde(default)]
    pub title: Option<String>,

    #[serde(default)]
    pub description: Option<String>,

    /// Status: "active" or "archived"
    #[serde(default)]
    pub status: Option<String>,

    /// Required for optimistic locking
    pub expected_version: i32,
}
